cores/esp8266/Schedule.cpp

```diff
@@ -60,19 +75,45 @@ bool schedule_function(std::function<void(void)> fn)
         sLast->mNext = item;
     }
     sLast = item;
+
+    if (repeat_us)
+        item->callNow.reset(repeat_us);
+
     return true;
 }
 
+IRAM_ATTR // called from ISR
+bool schedule_function(std::function<void(void)> fn)
+{
+    return schedule_function_us([&fn](){ fn(); return false; }, 0);
+}
+
 void run_scheduled_functions()
 {
-	scheduled_fn_t* rFirst = sFirst;
-	sFirst = NULL;
-	sLast  = NULL;
-    while (rFirst) {
-        scheduled_fn_t* item = rFirst;
-        rFirst = item->mNext;
-        item->mFunc();
-        item->mFunc = std::function<void(void)>();
-        recycle_fn(item);
+	scheduled_fn_t* lastRecurring = nullptr;
```

Comment:
I *think* this is IRQ safe, but we need one more pair of eyes.  My reasoning:

Enter `run_scheduled_function`
`toCall = sFirst`
<one more more `while` loops advance and call some things, but not done yet>
IRQ happens and is taken
IRQ calls `schedule_function`
The linked list is *appended* to
IRQ returns
At this point we have a new entry appended to the end of the list, but that's OK
<while continues, going thru the newly added entry at the end, too>

Reply:
> At this point we have a new entry appended to the end of the list, but that's OK

The new entries may not be executed on that loop if item is (was) the last one and interrupt added more, because `toCall` (= old item's next) would be already nullptr. But indeed that's OK.